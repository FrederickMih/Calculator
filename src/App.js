import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Math Magician Setup</h1>
      </header>
    </div>
  );
}

export default App;
