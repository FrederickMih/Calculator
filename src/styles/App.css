.app-main {
  display: flex;
  flex-direction: column;
  align-items: center;
  /* min-height: 100vh; */
  justify-content: space-around;
  align-content: center;
  /* margin-top: 1rem; */
}

.app {
  /* margin-top: 0; */
  max-width: 700px;
  /* margin-bottom: 1rem; */
  box-shadow: rgba(120, 9, 131, 2) 0 54px 55px,
    rgba(126, 16, 126, 0.12) 0 -12px 30px, rgba(218, 15, 236, 0.12) 0 4px 6px,
    rgba(71, 7, 80, 2) 0 12px 13px, rgba(73, 4, 70, 0.09) 0 -3px 5px;
}
