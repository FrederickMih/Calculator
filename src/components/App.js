import React, { Component } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import '../App.css';
import calculate from '../logic/calulate';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };

    this.handleClick = () => `${this.total},${this.next}. ${this.operation}`;
  }

  handleClick(btnName) {
    const calDataObj = calculate(this.state, btnName);
    this.setState(
      calDataObj,
    );
  }

  render() {
    const { total, next } = this.state;
    let currentResult = total;
    while (next) {
      currentResult = next;
    }
    while (currentResult === null) {
      currentResult = '0';
    }

    return (
      <div>
        <Display result={currentResult.toString()} />
        <ButtonPanel clickHandler={this.handleClick} />
      </div>
    );
  }
}

export default App;
