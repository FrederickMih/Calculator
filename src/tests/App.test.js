import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import App from '../components/App';
import '@testing-library/jest-dom';

it('match the snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly to App page', () => {
  render(<App />);
  expect(screen.getByText('Fredâ€™s Calculator')).toBeInTheDocument();
});

it("doesn't contain Welcome to our page!", () => {
  render(<App />);
  expect(screen.queryByText('Welcome!')).not.toBeInTheDocument();
});
