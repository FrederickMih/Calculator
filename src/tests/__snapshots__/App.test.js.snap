// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot 1`] = `
<div
  className="app-main"
>
  <div
    className="app"
  >
    <div
      className="app-display"
    >
      <h2
        style={
          Object {
            "fontSize": "3rem",
            "marginRight": "1rem",
            "textAlign": "right",
          }
        }
      >
        0
      </h2>
    </div>
    <div
      className="button-panel"
    >
      <div
        className="group-1 btn"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
            }
          }
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="group-2 btn"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
            }
          }
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="group-3 btn"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="group-4 btn"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="group-5 btn"
      >
        <button
          className="button exceptional-button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E0E0E0",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
  <h3
    className="fred"
  >
    Fred’s Calculator
  </h3>
</div>
`;
